app/models/transaction_type.rb

```diff
@@ -65,41 +63,36 @@ def to_param
     url
   end
 
-  def url_source
-    Maybe(default_translation_without_cache).name.or_else(nil).tap { |translation|
-      raise ArgumentError.new("Can not create URL for transaction type. Expected transaction type to have translation") if translation.nil?
-    }
-  end
-
-  def default_translation_without_cache
-    (translations.find { |translation| translation.locale == community.default_locale } || translations.first)
-  end
-
-  # TODO this should be done on service layer
-  def uniq_url
-    current_url = url_source.to_url
-
-    if new_record? || url != current_url
-      blacklist = ['new', 'all']
-      base_url = current_url
-      transaction_types = TransactionType.where(community_id: community_id)
-
-      i = 1
-      while blacklist.include?(current_url) || transaction_types.find { |tt| tt.url == current_url && tt.id != id }.present? do
-        current_url = "#{base_url}#{i}"
-        i += 1
-      end
-      self.url = current_url
-    end
-
-  end
-
   def display_name(locale)
-    TranslationCache.new(self, :translations).translate(locale, :name)
+    result = TranslationService::API::Api.translations
+      .get(community_id, {
+        translation_keys: [name_tr_key],
+        locales: [locale.to_s],
+        fallback_locale: community.default_locale
+      })
+
+    Maybe(result[:data].first)[:translation].or_else(nil).tap { |translation|
+      if translation.nil?
+        error_msg = "translation missing for key #{name_tr_key} and locale #{locale}"
+        ApplicationHelper.send_error_notification(error_msg, "TransactionType")
```

Comment:
yeah, this changed at some point to a direction where or_else could be nil and I should have changed this error handling too.